common/tier4_autoware_utils/CMakeLists.txt

```
@@ -9,6 +9,7 @@ find_package(Boost REQUIRED)
 ament_auto_add_library(tier4_autoware_utils SHARED
   src/tier4_autoware_utils.cpp
   src/geometry/boost_polygon_utils.cpp
+  src/ros/operation.cpp
```

I think it's good to make it more specific.
Or do you plan to add other operations to this file?

```suggestion
  src/ros/quaternion_operation.cpp
```

---

I think we can also add +, -, * of geometry_msgs's Pose and Point, whose operation is widely used in autoware.universe.

---

How about `msg_operation` or `geometry_msg_operation`?

---

Sounds good to me.
https://github.com/autowarefoundation/autoware.universe/pull/2244/commits/6a310b7138c36c1d011a4458e1bf31dbc770d141

---

Will this msg_operation support mathematical operations for other messages? like twist_a + twist_b

---

Of course we can add them if we want.
In this PR, I just copied from motion_common, which is currently required implementation for https://github.com/autowarefoundation/autoware.universe/pull/2244#issuecomment-1306697597